gammapy/datasets/utils.py

```diff
@@ -203,3 +214,165 @@ def split_dataset(dataset, width, margin, split_template_models=True):
+    )
+    geom_exposure = geom.to_image().to_cube([energy_axis_true])
+
+    # create dataset and run data reduction / irfs interpolation
+    dataset = MapDataset.from_geoms(
+        geom,
+        geom_exposure=geom_exposure,
+        geom_psf=geom_psf,
+        geom_edisp=geom_edisp,
+        name=name,
+    )
+
+    selection = ["exposure", "edisp", "psf"]
+    if obs.events:
+        selection.append("counts")
+    if obs.bkg:
+        selection.append("background")
+
+    maker = MapDatasetMaker(selection=selection)
+    dataset = maker.run(dataset, obs)
+
+    if "background" not in selection:
+        dataset.background = Map.from_geom(geom, data=0.0)
+
+    return dataset
+
+
+def add_fermi_isotropic_model(dataset, isotropic_filepath, as_bkg=True):
+    """Add Fermi-LAT isotropic model to dataset
+
+
+    Parameters
+    ----------
+    dataset : `~gammapy.datasets.MapDataset`
+        Map dataset.
+    isotropic_filepath : str
+        Isotropic file path.
+    as_bkg : bool, optional
+        If True set the isotropic as background model instead of a SkyModel.
+        The default is True.
+    """
+
+    if dataset.models is None:
+        models = Models()
+
+    diffuse_iso = create_fermi_isotropic_diffuse_model(
+        filename=isotropic_filepath, interp_kwargs={"extrapolate": True}
+    )
+    diffuse_iso.apply_irf["edisp"] = False
+
+    dataset.models = models + [diffuse_iso]
+
+    if as_bkg:
+        dataset.background = dataset.npred()
+        dataset.models = models + [FoVBackgroundModel(dataset_name=dataset.name)]
+
+
+def create_map_dataset_from_fermi_files(
+    path,
+    isotropic_filename,
+    events_filename="ft1_00.fits",
+    counts_filename="ccube_00.fits",
+    exposure_filename="bexpmap_roi_00.fits",
+    psf_filename="psf_00.fits",
+    drm_filename="drm_00.fits",
+    geom=None,
+    energy_axis_true=None,
+    name=None,
+    isotropic_as_bkg=True,
+):
+    """
+
+
+    Parameters
+    ----------
+    path : TYPE
+        DESCRIPTION.
+    isotropic_filename : str
+        Isotropic file name.
+    events_filename : str
+        Events filename. Default is ft1_00.fits
+    counts_filename : str, optional
+        Counts filename. Default is ccube_00.fits
+    exposure_filename : str, optional
+        exposure filename. Default is bexpmap_roi_00.fits
+    psf_filename : str, optional
+        PSF filename. Default is psf_00.fits
+    drm_filename : str, optional
+        DRM filename. Default is drm_00.fits
+    obs : `~gammapy.data.Observation`
+        Observation with DL4 IRFs
+    geom : `~gammapy.maps.WcsGeom`, optional
+        Output dataset maps geometry. The default is None, and it is derived from IRFS
+    energy_axis_true : `~gammapy.maps.MapAxis`, optional
+        True energy axis used for IRF maps. The default is None, and it is derived from IRFS
+    name : str, optional
+        Dataset name. The default is None, and the name is randomly generated.
+    isotropic_as_bkg : bool, optional
```

Comment:
This is another interesting point.
Should the iso and galdiff considered as bkg or not. 

The way I tried it, everything was loaded as a Sky model. But then the issue is that there bkg is empty and for example for the peek() the excess attribute doesn't work because there is no background.

It is not critical because the rest of the fitting & plotting functions but it introduces a difference between the IACT and Fermi Dataset if Fermi has empty bkg.
Both options are valid in the end, just a question of choosing the default behavior. 
In my opinion if `iso` is default in the background, `galdiff` should be as well no?

Reply:
I am not sure there needs to be a convention for this. For IACTs we typically consider background as instrumental (and not astrophysical). If necessary the same logic could apply to Fermi, which would have no background set.

Reply:
agreed
but If the bkg is always None. Then some functions don't work like the excess in the `peek()` functions or when printing the `Dataset`. 
Maybe be not a big issue, but the first time I swa these errors it took me sometime to understand why.